mono/metadata/mono-hash.c

```diff
@@ -249,8 +249,8 @@ rehash (MonoGHashTable *hash)
 guint
 mono_g_hash_table_size (MonoGHashTable *hash)
 {
-	g_return_val_if_fail (hash != NULL, 0);
```

Comment:
I like pushing the null check to the callees, I think it brings consistency with what we expect of C APIs.
What I'm on the fence if wether we want to do it like this, expanding g_return_val_if_fail.

Reply:
I think these are internal apis, so they should just assert.